mne/datasets/config.py

```diff
@@ -87,7 +87,7 @@
 # respective repos, and make a new release of the dataset on GitHub. Then
 # update the checksum in `mne/data/dataset_checksums.txt` and change version
 # here:                  ↓↓↓↓↓         ↓↓↓
-RELEASES = dict(testing='0.123', misc='0.18')
+RELEASES = dict(testing='0.123', misc='0.22')
```

Comment:
The `misc` dataset was never updated since before pooch went in https://github.com/mne-tools/mne-python/blob/034ff3f03ae1ae01a6fe52e3e91e10682c977aff/mne/datasets/utils.py#L257

So I updated it here. This results in a few examples and tests breaking because it looks like `sample_ecog.mat` is replaced by `sample_ecog_ieeg.fif`. I point out where I updated in those files.